compute/instances.go

```diff
@@ -249,13 +249,24 @@ func (input *CreateInstanceInput) toAPI() map[string]interface{} {
 		Far:    input.LocalityFar,
 	}
 	result["locality"] = locality
+
 	for key, value := range input.Tags {
 		result[fmt.Sprintf("tag.%s", key)] = value
 	}
 
 	// Deliberately clobber any user-specified Tags with the attributes from the
 	// CNS struct.
+	//
+	// NOTE(justinwr): CNSTagServices needs to be tagged so that our instance
+	// can be provisioned with CNS. No other CNS tags will be handled at this
+	// time.
 	input.CNS.toTags(result)
+	for key, value := range result {
+		if key == CNSTagServices {
+			result[fmt.Sprintf("tag.%s", key)] = value
```

Comment:
I can definitely clean this up a bit and rely more on `CNSTagServices`. That is the only tag we need to prepend with `tag.` here, anything else slips through.